### ISSUE TITLE
One more patch to have a working WIFI

### ISSUE THREAD

**Issue by flipphos** [syscl/XPS9350-macOS#82]
Many thanks for the great work! It makes my installation on XPS 13 9350 quite straight forward.
However, WIFI was not working, until I added one more path according to the article here:
http://www.insanelymac.com/forum/topic/312759-fix-macos-sierra-dp1-bcm94532z-wifi/

Basically, what I did was:
Kext: AirPortBrcm4360 
Find: 81F952AA00007529 
Replace: 81F952AA00006690

After reboot, WIFI works fine with both 2.4G and 5G.

I hope you could add this patch into your new release, so other people could benefit a bit from what I learnt. Thanks!

Cheers,
HG

**Comment by aizigao** [comment]
xps9350 use 1820A which has bugs，see 
#22

**Comment by ZombieTheBest** [comment]
@aizigao I'm experiencing WiFi problems with DW1830 too after StandBy.

**Comment by flipphos** [comment]
@aizigao I have replaced 1820A with DW1560, which I am currently using.
I suggest @syscl add one more patch in the my post into the next release, which I used to make mine WIFI working.

**Comment by aizigao** [comment]
@flipphos 1560 https://rossfisher.me/2016/12/how-to-bcm94352z-for-hackintosh-including-5ghz-patch-handoff-and-hotspot-macos-sierra/

**Comment by aizigao** [comment]
@flipphos  nice jobs 👍

**Comment by ZombieTheBest** [comment]
@flipphos The patch you provided fixed my issue

**Comment by syscl** [comment]
@ZombieTheBest @aizigao If you have DW1830, then this fcvo patch doesn't require. I have some ideas about the Wi-Fi slow response issue, please give this a try(disable reg-ltrovr``` by commenting out the following code in ```Deploy.sh``` like this:
```sh
#   _tidy_exec "patch_acpi DSDT syscl "syscl_ltrovr"" "Inject reg-ltrovr for IOPCIFamily::setLatencyTolerance setting ltrOffset for PCI devices successfully (c) syscl"
```

As a note, remove ```fcvo``` patch in your config as well.

syscl

**Comment by ZombieTheBest** [comment]
@syscl This may have fixed the issue, need more testing

**Comment by ZombieTheBest** [comment]
@syscl Nope. Issue was not fixed. Sometimes Wifi is very slow after sleep (don't have to be a "long" one). Connecting to another AP solves the issue. The problem is not caused by reg-ltovr

**Comment by illladelph** [comment]
I am having this problem now and I don't think I was having it previously. Ever since I got a new router (Netgear R7000) I have had seriously slow speeds after sleep. Not sure if its related to updating my OSX/syscl patch, or the new router but I am seeing 100+mbit speedtests on a clean boot but after sleep for any amount of time, it drops substantially to 3mbit or less. Occaisionally it drops to <.5 mbit and I'm clearly losing packets as well. Hoping someone has this issue as well or any suggestions.

**Comment by furrybanana** [comment]
@illladelph is this with a DW1560 ?

try this https://github.com/syscl/XPS9350-macOS/issues/100#issue-229663705 :
Quick fix: if you wake your XPS from sleep and wifi is slow, do a software sleep (top left corner -> Apple -> Sleep), wait till Power button goes dark. Press Power button -> Wifi is fast again.
/edit2: this sometimes leaves you without working audio. Re-Sleep fixed it for me.

**Comment by illladelph** [comment]
@furrybanana sorry yea its the 1830. I've been having issues for months now with either wifi or BT. Previously the BT would always show not available after sleep. I recently updated to 10.12.5 and re-deployed and the wifi/bt was seemingly better. But after sleep it is hopelessly slow. I have tried doing the soft sleep like you advised and no change in speeds. One thing I do notice is on speedtests, the ping goes from ~20ms on clean boot to ~1-200+ms after sleep. So there is clearly some kind of bug in the wifi resuming from sleep. Since the update though I have not had any BT issues and so far I haven't noticed a loss of sound from any sleep whereas before I would lose audio randomly and had to open system prefs-sound-input to get it to come back.

**Comment by illladelph** [comment]
Is anyone else having this issue with the 1830? Should I switch to the 1560? I sleep my computer usually 5-10 times a day so this is getting pretty frustrating and I don't believe it was an issue in previous releases of this patch

**Comment by kultigspritzig** [comment]
@illladelph I have exact same issue with 1830. It's pretty much unusable at his point.

**Comment by mingiryu** [comment]
#81 DW1830 with macOS 10.12.3. Wifi works normally after sleep.

**Comment by mingiryu** [comment]
@illladelph Same issue with DW1830 on macOS 10.12.3

**Comment by furrybanana** [comment]
finally, I had the time to switch to the 1560. thank you @flipphos 

This was a huge downer for using the hackintosh with 1830. 

I recommend getting the 1560 for using with macOS.

**Comment by rpruden** [comment]
I seem to have this problem too. I max out at about 15Mbps after sleep. Also when running a ping to google.com, I was getting consistent 40ms-60ms before sleep and then after sleep, I get 70ms-150ms. I had ordered the 1830 because I thought it was more widely supported :-(

Anyone had luck with getting handoff working on either the 1830 or 1560? I have it working one way only. It will send pages and clipboard to my IOS device, but it will not receive the other way around.

**Comment by mingiryu** [comment]
@rpruden I think handoff feature uses bluetooth like the airdrop feature. It's very likely that it's the DW1830 card that is causing both wifi and bluetooth problem because airdrop is sometimes broken after sleep. I had to turn off and on the bluetooth in order to make it work again. 
The wifi is a well known issue now. You can manually put the device back to sleep through the left corner menu to get the normal wifi speed, but it's annoying when you have sleep setup as 5 min inactivity or something.

**Comment by rpruden** [comment]
@mingiryu Thanks for the response. My handoff issues don't only occur after sleep though. My handoff will send clipboard items and safari tabs TO my iPhone, but it doesn't work from my other devices TO this laptop for some reason. I have tried using the DW1560 and I'm currently using the DW1830.

I had ordered the DW1830 in hopes that it would resolve this handoff issue but the behavior is the same. Which card do people consider to be offering the more complete functionality at this point? It seems to me that there are mixed responses on this point.

**Comment by mingiryu** [comment]
@rpruden based on furrybanana's post, DW1560 seems to be the best option at the moment. DW1830 sleep issue.

**Comment by rpruden** [comment]
Cool. Thanks for the advice. I will swap this card back out and hang on to the DW1830 for a rainy day or something.

**Comment by illladelph** [comment]
I can confirm that replacing the DW1830 w the DW1560 is a good fix for now. I was unable to get it working by re-deploying syscl's patch so I had to follow this:

First: Installing Kexts to get Wifi+BT working:

1. Download the latest release from https://bitbucket.org/RehabMan/os-x-fake-pci-id/downloads

2. Download the latest release from https://bitbucket.org/RehabMan/os-x-brcmpatchram/downloads

3. Copy BrcmFirmwareData.kext, BrcmPatchRAM2.kext, FakePCIID_Broadcom_WiFi.kext and FakePCIID.kext to your EFI partition (EFI/Clover/kexts/10.12/) —  If you have not yet updated Clover boot loader (such as using UniBeast which has the old Clover install), copy to EFI/Clover/kexts/Other)

4. Open Clover Configurator and in the Kernel and Kext patches, add:

    Name:    AirPortBrcm4360
    Find:    81F952AA00007529 
    Replace: 81F952AA00006690
    Comment: Brcm4360 Sierra fvco init

5. Restart

**Comment by louwe** [comment]
Is anyone else getting this issue? I'm using DW1560. WiFI works great for the most part, but bluetooth would stop working after sleep and the only way to get it back is through a reboot.

**Comment by illladelph** [comment]
lol I was about to edit my post to say the same thing. Bluetooth is lost on resume from sleep 5g working tho :/

**Comment by louwe** [comment]
@illladelph I actually have diagnosed this problem more and here's what I noticed.

After a deep sleep (i.e. power button LED turns off), upon wake up, bluetooth will no longer be available, however WiFi still works.
The next deep sleep will then disable WiFi and bring a new issue. The laptop itself will now lose the ability to sleep at all.

Bluetooth is a minor annoyance, but WiFi not working as well as laptop not being able to sleep at all are huge deal breakers. I'm considering swapping my DW1820A back cause though the card only worked in Windows, at least I didn't have to deal with sleep problems.

**Comment by louwe** [comment]
UPDATE:
I may have found the solution to it.
A thought came to me that the newly-installed card may need some "priming" to sleep properly.
So, I rebooted into Windows 10 and did several deep sleep/wake-up cycles and rebooted back to Mac OS Sierra.
To my delight, the priming seems to be the key I needed to solve the problem.

I'm still holding my final judgment though. I will observe to see if the issue crops back up over the next several days.
Stay tuned!

**Comment by louwe** [comment]
UPDATE:
Sleep issue still exists somewhat infrequently. It seems like a solid way to reproduce it is by having a bluetooth device connected while trying to sleep.

Knowing that, I disabled bluetooth in the BIOS and now I have zero sleep issues.
Seems like the Bluetooth card not being able to wake up also directly affects the WiFi card.
Now that it has been disabled though, it can no longer hamper the WiFi card.
That being said, it would still be nice to be able to use the Bluetooth card without fear of the WiFi card being disabled and sleep issues cropping up.

**Comment by louwe** [comment]
UPDATE:
I've resolved the issue with some help from folks at tonymacx86.com.

I now have a fully functional WiFi and Bluetooth without any sleep issues whatsoever.
The root cause turns out to be unused USB ports that were still configured.

Since XPS 13 9350 i7-6560U (not sure about other models) only has 2x USB2/3 ports + 3 internal ports, it didn't need the extra 7 ports or so configured by the deploy script listed under SSDT-XHC.aml file (i.e. HS09, USR1, USR2, SS03 - SS06).

I think this should be added together with the WiFi patch @flipphos mentioned for a glitch-free sleep install.

**Comment by illladelph** [comment]
Awesome work. Could you provide info on how to resolve this?

**Comment by syscl** [comment]
Hi @flipphos @ZombieTheBest @aizigao @illladelph @furrybanana please try this aml then report!
[SSDT-XHC.aml.zip](https://github.com/syscl/XPS9350-macOS/files/1193268/SSDT-XHC.aml.zip)

Thanks in advance,
syscl

**Comment by flipphos** [comment]
Alright. I have done some experiment with the new SSDT-XHC.aml from syscl's email. Before start, I also updated clover to the latest 4128.

- For convenience, I adjusted the time for sleep to 5 min, hooked a Magic mouse gen. 2 with the laptop via bluetooth.

Leave the 9350 till the power LED turns off, while doing some other things. Turn the laptop on again by pressing the power button, I had to press any key to wake up the screen, or the screen turns black again. The wifi and bluetooth seem to all work fine after each wakeup, as I could see emails coming in and move the cursor by the mouse. I repeated this loop for 6 times. No problems so far.

I just rebooted the laptop because I updated some files in the clover folder such as lilu and its plugins, and the touchpad driver. I have had the sleep/wake up twice now, and I am going to leave it overnight and see what happens tomorrow morning. I’ll update you tomorrow.

By the way, @syscl is it possible to include the AppleALC plugin to work with lilu to replace the current kext AppleHDA_ALC256.kext, ALC256 (3246) is listed as supported by the plugin?

Cheers,
flipphos

**Comment by illladelph** [comment]
Updated clover, downloaded new patch and replaced aml file and I am getting stuck at installing voodoops2 kext. After reboots still getting stuck

**Comment by syscl** [comment]
@flipphos sure, I will try to merge it.

Thank you for detailed report!

All the best,
syscl

**Comment by jerryzj** [comment]
Hi,@syscl

Thanks for your amazing project first.

After deploy your latest script(uploaded 4 hours ago), I think the slow WiFi issue still exist.
After sleeping(Power LED close), I do speedtest and I got this.
![slow](https://user-images.githubusercontent.com/11689182/28901432-a85745de-782a-11e7-8a04-0d25ee9a85bb.png)
Therefore, I sleep(Power LED still on) again and this is what I get.
![fast](https://user-images.githubusercontent.com/11689182/28901435-b24b5364-782a-11e7-816c-2e81a266388a.png)

My hardware configuration
XPS13 9350 i5 8G
DW1830

For your reference.

Best,
Jerry

**Comment by syscl** [comment]
@jerryzj Thanks for your feedback.

So twice sleep resolve the slow Wi-Fi issue?

syscl

**Comment by jerryzj** [comment]
@syscl Yes.

Jerry

**Comment by illladelph** [comment]
I'm still having issues after sleep with no 5G and lost BT

**Comment by louwe** [comment]
So, with the new USB configuration, I've had solid sleeps without any issues at all for about a whole week now with the DW1560.

The key here though, is that I've disabled BT in the BIOS (I do not use BT at all).
In my experience, when BT is enabled, on certain sleeps, there is a chance that first the BT card will stop working (WiFi still works at this point).
But then, the next sleep WILL disable both your BT AND your WiFi.

Since I do not use BT and has no plans of using it in the future, I can live just fine with it being disabled. My advice is, if you do not use or can live without it, disable it in the BIOS and your life will be 1000% better.

**Comment by flipphos** [comment]
My update here:

I have tested wifi and BT after sleep/wake over the last two nights after applying the new aml file. Both wifi and BT (hooked a Magic mouse gen 2) worked fine. However, I didn’t tested intensively, just left it to sleep (power LED turned off) and then woke it up by pressing the power button, and tested wifi and moved the mouse (BT) to confirm both were working. In my 7 - 8 loops tests over 2 nights, both worked fine. 

I did remember that I experienced BT lost after sleep when I was using a Magic touchpad gen 1. Turn the BT off and on could get the BT back working again.

As I said, I haven’t tested both wifi and BT intensively enough, though, the sleep shall work fine for me for now.

Thank you @syscl for this great project!

Cheers,
flipphos

**Comment by se1ya** [comment]
if have the same problem. installed is the 1830.
befor: ping google.com 20-30
after: ping google.com 200-2000 (with package losings)

i disabled the bluetooth in my bios, but it had no effects.

at the moment this issue is the biggest for a daily using :(
<img width="478" alt="screenshot 2017-08-10 18 08 35" src="https://user-images.githubusercontent.com/16716560/29181998-7434d914-7dfd-11e7-8d31-9984be4fd9a1.png">

**Comment by ZombieTheBest** [comment]
Hi @syscl @flipphos @louwe Swapped today DW1830 with DW1560. WiFi slow issue was still present, so I have updated the SSDT-XHC.aml file. Right now I'm still testing but it seems to work fine (both WiFi and Bluetooth). The only thing I'm noticing (not sure if it's caused by bluetooth or by the low quality headphones), headphones sometimes lose bluetooth signal for some seconds, especially if I cover bluetooth card with my hands when typing. It seems like there's a low signal. Anyone else experiencing this?

**Comment by louwe** [comment]
@ZombieTheBest I suppose it's the price I pay for having solid WiFi. Oh well, I'd rather have solid WiFi over being able to use Bluetooth.

**Comment by ZombieTheBest** [comment]
Ok thank @louwe. For now in my case bt and wifi seems to work both. I will test more 👍

**Comment by syscl** [comment]
Hi @ZombieTheBest , DW1560 solve BT+WiFi issue?

syscl

**Comment by ZombieTheBest** [comment]
@syscl I swapped the card yesterday. For now both BT and WiFi are working
after sleep. The only problem I'm encountering is low Bluetooth signal (not
sure if it's caused by card or low quality headphones, will let you know
next week as I'm waiting to receive Apple AirPods). If I cover BT card with
my hands (while typing for example) I tend to loose audio signal for some
seconds, but not always.

2017-08-12 5:10 GMT+02:00 Zhou Yating <notifications@github.com>:

**Comment by illladelph** [comment]
Sent in my XPS for repair they replaced mobo/heatsink/fan and now that I have it back I have been running the DW1560 for two days with a fresh dual boot install of Sierra/Win10. So far so good. I ran into all kinds of other unrelated issues with installation for some reason but I got it all sorted. Wifi/BT have been working for me 5G/Handoff/Continuity/Airdrop after sleep and after deep sleep. No issues with BT signal like zombie. I do see a loss in signal at around 10-15ft away with my hands covering the keyboard but at that distance only. 

Strangely while everything is working for me now on Mac/Win side my bios and clover are losing keystrokes/ making it very difficult to navigate. It is extremely strange and I reset bios but still I have to press several times with the arrow key to get it to register. I doubt seriously that it has anything to do with this patch as it only does it in BIOS and Clover bootloader, but strange regardless

**Comment by se1ya** [comment]
@illladelph  i have the same with the slow ui in clover from the beginning. it was with different versions of clover, i think its not a problem of clover

**Comment by se1ya** [comment]
hey, can someone test this:
- disable WIFI in the status bar symbol
- sleep ur machine
- wake up 
- start the WIFI again at the status bar now

i dont have the slow down at the WIFI connection. 

at the moment its the best solution for me, its only 2 clicks and no slow WIFI

**Comment by illladelph** [comment]
Well updated last night to 10.12.6 and now after deep sleep I have lost BT like before...

**Comment by syscl** [comment]
BT issue, hmmmm, I believe this issue is somewhat located in BrcmPatchRAM...

syscl

**Comment by ZombieTheBest** [comment]
I got Apple AirPods. I can confirm BT issue persists when on 2.4Ghz WiFi with DW1560. Disabling WiFi helps a lot, but BT Signal is still a little bit weak. After some days I can say that there are still the slow WiFi after sleep issue with DW1560, even if it's not as bad as DW1830. 
Yesterday I woke up my laptop from sleep and copied a file from my NAS over 2.4Ghz WiFi and I was getting 1.2MB/s. I did a reboot and got 8MB/s, so the problem is still present.

Any way to fix this? (BT is very important for me with AirPods)

**Comment by syscl** [comment]
@ZombieTheBest

I am on my flight back to US, so could you please try to comment out this line in deploy.sh and see if this fix your Wi-Fi sleep issue?
```sh
 _tidy_exec "patch_acpi DSDT syscl "syscl_ltrovr"" "Inject reg-ltrovr for IOPCIFamily::setLatencyTolerance setting ltrOffset for PCI devices successfully (c) syscl"
```

Thank you in advance,
syscl

**Comment by syscl** [comment]
@ZombieTheBest

I am on my flight back to US, so could you please try to comment out this line in deploy.sh and see if this fix your Wi-Fi sleep issue?
```sh
 _tidy_exec "patch_acpi DSDT syscl "syscl_ltrovr"" "Inject reg-ltrovr for IOPCIFamily::setLatencyTolerance setting ltrOffset for PCI devices successfully (c) syscl"
```

Thank you in advance,
syscl

**Comment by jerryzj** [comment]
@syscl maybe I can test this now.

**Comment by jerryzj** [comment]
@syscl 目前看起來沒有改善
我只有改完之後直接重跑Deploy.sh
是不是我有漏了什麼步驟？

**Comment by se1ya** [comment]
hey i tested it. i comment out the line but no effects at my system with DW1830
<img width="350" alt="screenshot 2017-08-17 08 09 12" src="https://user-images.githubusercontent.com/16716560/29398233-bdf6843c-8323-11e7-96a6-2fecb49c2f14.png">

before i had stabile 20-30ms

**Comment by ZombieTheBest** [comment]
Hi @syscl 
Thank you very much for all your time. 
I commented out the line and launched Deploy.sh again with a reboot.

Wi-Fi sleep issue still present.

Before Sleep:
<img width="537" alt="schermata 2017-08-17 alle 10 07 25" src="https://user-images.githubusercontent.com/22716366/29402144-2de11068-8334-11e7-867a-47db84470908.png">

After Sleep:
<img width="539" alt="after" src="https://user-images.githubusercontent.com/22716366/29402339-e4639e32-8334-11e7-9e03-5caf025fb1fa.png">

I'm gonna revert back the changes made. @bozma88 has DW1560 on XPS9360, but I guess he is not experiencing this WiFi slow issue on his laptop (never seen any post on his XPS tutorial post on tonymac). Are we missing something, cause I have WiFi/BT problems on both cards. I saw I bozma's kext folder a kext named "FacePCIID_Broadcom_WiFi". Is that for DW1560?


Pps: don't know if it's my impression but it seems like that slow-WiFi occurs in Upload side, download side is of course limited by the very bad upload.

**Comment by syscl** [comment]
@ZombieTheBest @se1ya @jerryzj  Thank you for the feedback.

So ```reg-ltrovr``` is not the reason that causes slow Wi-Fi.

As ZombieTheBest's question, I've already included FakePCIID_Broadcom_WiFi even in my first initial commit of this repo. 

Could you please try to remove FakePCIID_Broadcom_WiFi under Clover/kexts to see if this solve your slow Wi-Fi issue?

syscl

**Comment by ZombieTheBest** [comment]
@syscl Nothing changes when removing FakePCIID_Broadcom_WiFi for me

**Comment by illladelph** [comment]
W 1560 my problem still persists. Switching from 1830 to 1560 doesn't seem to have helped at all on 10.12.6. Before I updated it was working great with 1560 but after update I am losing BT/Wifi after sleep. I also cannot get my sleep settings to work. The computer display ALWAYS turns off after a few min and goes to sleep causing loss of BT/wifi or both. I have changed my power settings to: never turn off display, checked prevent computer from sleeping while display is off, unchecked all the rest, yet it still turns off the display after a few min and sleeps.

**Comment by louwe** [comment]
@illladelph I also have that issue you have with losing BT and then WiFi. But, since I do not use BT, I simply disabled it in the BIOS. Once disabled, WiFi + sleep works so solidly that I could go on for 2 weeks without rebooting.

**Comment by illladelph** [comment]
@louwe that seems like a good temporary situation but unfortunately for me I rely heavily on BT. I use airdrop multiple times a day to get photos off my phone so BT is quite important to me.

**Comment by ZombieTheBest** [comment]
I've asked for support on tonymac forum regarding WiFi+BT issue and rehabman is trying to help me. First thing to do is fix up everything he says I guess eheh :D 

I leave you a link if you wanna follow!
https://www.tonymacx86.com/threads/dell-dw1560-dw1830-wifi-bt-issue-on-xps9350.230858/

**Comment by ZombieTheBest** [comment]
#145 Try this and let me know!

**Comment by C9HDN** [comment]
If this patch isn't working for you, change "AirPortBrcm4360" to "com.apple.driver.AirPort.Brcm4360." so that the patch looks like this.

Kext: com.apple.driver.AirPort.Brcm4360
Find: 81F952AA00007529
Replace: 81F952AA00006690

Don't forget the dot after AirPort. 

https://www.tonymacx86.com/threads/solved-bcm94352z-does-not-work-wifi-in-sierra.240308/
